### PR TITLE
Update examples and documentation to 0.7

### PR DESCRIPTION
**Pull request by Erk-**
This pull request updates all examples to 0.7, it also moves the
permission calculator into utils.

### CODE REVIEW

File: src/chapter_1_crates/section_3_gateway.md [hunk 1]
```diff
@@ -14,14 +14,16 @@ usually want to spawn a task to bring it up such that you can begin to receive
 tasks as soon as they arrive.
 
 ```rust,no_run
+# use std::sync::Arc;
 # use futures::StreamExt;
 # use twilight_gateway::{Cluster, Intents};
-#
+
```
Comment (7596ff) [inline src/chapter_1_crates/section_3_gateway.md]: I believe this change can be undone

File: src/chapter_1_crates/section_7_first_party/section_4_util.md [hunk 2]
```diff
@@ -51,11 +51,17 @@ Retrieve the timestamp of a snowflake in milliseconds from the Unix epoch as a
...
 let timestamp = user.timestamp();
 # }
 ```
 
+### Permission Calculator
+
+The `permission-calculator` feature is used for calculating the permissions
+of a member in a channel, taking into account its roles and permission
+overwrites.
+
```
Comment (7596ff) [inline src/chapter_1_crates/section_7_first_party/section_4_util.md:101]: we should add a note about the builders as well

File: tests/Cargo.toml [hunk 3]
```diff
@@ -3,22 +3,26 @@ name = "twilight-book"
...
+twilight-mention = { version = "0.7" }
+twilight-standby = { version = "0.7" }
+twilight-util = { version = "0.7", features = ["full"] }
+tracing-subscriber = "0.3"
 
 [build-dependencies]
 skeptic = "0.13.5"
+
+[dev-dependencies]
+skeptic = "0.13"
```
Comment (7596ff) [inline tests/Cargo.toml]: is this meant to be duplicated?

**Review by 7596ff — CHANGES_REQUESTED** [review]

File: tests/Cargo.toml (hunk 3 above)
Comment (Erk-) [reply tests/Cargo.toml]: This is how skeptic recommends it upstream, though I can check if it is really needed.

Comment (7596ff) [reply tests/Cargo.toml]: Oh, I didn't notice one was build dependencies and one was dev.

Comment (7596ff) [inline tests/Cargo.toml]:
```suggestion
skeptic = "0.13"

```

**Review by 7596ff — APPROVED** [review]

**Pull request merged**